### ISSUE TITLE
Try unify the precompile error rule

### ISSUE THREAD

**Issue by AsceticBear** [darwinia-network/darwinia-common#1217]
After #1214 finished. The application side is able to capture more information about the dispatch error. It also introduced another issue it's easy to mix up the right approach to decode the captured error message for beginners. In most cases, such message `helper.revert("error")` can do ASCII convert to get the readable content. However, for the dispatch error message, you have to do another scale decode. That might be confused sometimes.

After discussing with Aki and Echo, We have decided to choose a similar design with solidity, and return the ethabi encode of the result, like `Error(string)` and wish Remix or Metamask could decode the message binary automatically.

The expected result after this:

![image](https://user-images.githubusercontent.com/11801722/168578295-2d72090b-8577-4b54-a551-2aee062e6f6d.png)

Instead of `08c379.....`, show something readable like `xxx decode failed.`

**Comment by AsceticBear** [comment]
I found this helpful QA https://ethereum.stackexchange.com/questions/76653/how-to-decode-error-response-from-contact
